### DISCUSSION TITLE
Command Line to control sizing / placement of a window?

### DISCUSSION THREAD

**Discussion by bric3** [rxhanson/Rectangle#1484] [General]
Is there a way to apply a configuration, e.g. "Last Two Third" to a specific window from the command line ? Or would it be possible via a script ?

I suppose getting the right window can be an issue (from command line), maybe having a way to filter by window title could work. And then I suppose this gives a window identifier that could be used to tell Rectangle to apply a configuration.

Why I would want that?
I'm thinking about making a plugin for JetBrains IDEs that could tell Rectangle to place the current IDE window in a certain way, and following certain criteria, maybe something to do in a single project window, but not the others.
Also the IDE is ignored in Rectangle due to conflicting shortcuts.

**Comment by rxhanson** [comment]
You can execute Rectangle actions by URL:

https://github.com/rxhanson/Rectangle?tab=readme-ov-file#execute-an-action-by-url

**Reply by bric3** [reply]
This is great, I'll be playing with a bit more !

**Reply by bric3** [reply]
@rxhanson Is it possible to Ignore / Unignore the current app, from the an "action" URL ? (It's not mentioned in the list, and I don't think I saw that in the code but maybe I missed something there.)

**Reply by bric3** [reply]
I crafted the plugin for IntelliJ IDEA this week-end, now for the plugin icon I wondered if I could use the following asset.

https://github.com/rxhanson/Rectangle/blob/eee0492541e8b7b614e9723dc14b7d6c1a37d057/Rectangle/Assets.xcassets/AppIcon.appiconset/mac512pts2x.png

By any chance do you have an SVG version ?

Otherwise I was thinking about making a simple svg.
![rectangle-icon-outline](https://github.com/user-attachments/assets/c7dfd986-12bf-4810-a277-719217ee217b)

**Reply by rxhanson** [reply]
I use Pixelmator Pro for image editing, and exported it, but it's not quite right.

![rectangle](https://github.com/user-attachments/assets/87e2557e-32fe-4a6f-be10-69013a845d84)

You could probably easily modify that to get it whipped into shape.

Even better, here's the original contribution with a link to the figma that is still up (after about 4 years :).
https://github.com/rxhanson/Rectangle/issues/266

**Reply by rxhanson** [reply]
Also, I don't have the bandwidth at the moment but would accept a pull request for url ignore/unignore if anyone's interested in contributing.

**Reply by bric3** [reply]
I don't know Swift or even where to start at this time. So no promise.

About the SVG the one posted seems quite good already, but here's the one I extracted from Figma (for some reason though the browser don't render it as well as inkscape or figma):

![rectangle-icon](https://github.com/user-attachments/assets/740d0a6e-1a9f-482b-8014-c25c04542432)

Should I include the MIT license header in this file ?

----

It renders well in JetBrains IDE (I'll open source the code when ready):

![image](https://github.com/user-attachments/assets/b19861ed-f865-46d7-86b0-e654df5070b9)

**Reply by bric3** [reply]
> Also, I don't have the bandwidth at the moment but would accept a pull request for url ignore/unignore if anyone's interested in contributing.

@rxhanson Please review with care, as I'm quite a newbie in swift. https://github.com/rxhanson/Rectangle/pull/1505

**Comment by bric3** [comment]
@rxhanson https://plugins.jetbrains.com/plugin/25677
